proposals/lip-0045.md

```diff
@@ -1583,7 +1583,7 @@ If `chainInfos` is not empty, then check that:
   * let `totalWeight` be the sum of the `bftWeight` property of every element in `activeValidators`. Then `totalWeight` has to be less than or equal to `MAX_UINT64`;
   * check that `totalWeight//3 + 1 <= certificateThreshold <= totalWeight`, where `//` indicates integer division;
   * check that the corresponding `validatorsHash` stored in `mainchainInfo.chainData.lastCertificate.validatorsHash` matches with the value computed from `activeValidators` and `certificateThreshold`.
-* Each entry `stateAccount` in `terminatedStateAccounts` has a unique `stateAccount.chainID` and `terminatedStateAccounts` is ordered lexicographically by `stateAccount.chainID`. Furthermore for each entry it holds `stateAccount.chainID != getMainchainID()` and `stateAccount.chainID != ownChainAccount.chainID`.
+* Each entry `stateAccount` in `terminatedStateAccounts` has a unique `stateAccount.chainID` and `terminatedStateAccounts` is ordered lexicographically by `stateAccount.chainID`. Furthermore for each entry it holds `stateAccount.chainID != getMainchainID()`, `stateAccount.chainID != ownChainAccount.chainID` and `stateAccount.chainId[0] == getMainchainID()[0]`.
```

Comment:
Is `ownChainAccount` actually defined already at this point? Shouldn't we check against `OWN_CHAIN_ID` and initialize the `ownChainAccount` in the genesis state finalization?

Reply:
Good point. Changed the check.